packages/vm/lib/runTx.ts

```diff
@@ -245,16 +255,31 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     }
   }
 
+  let gasPrice
+  let inclusionFeePerGas
+  if (this._common.isActivatedEIP(1559)) {
+    const baseFee = block.header.baseFeePerGas
+    const EIP1559Data = tx.getEIP1559Data()
+    inclusionFeePerGas = BN.min(
+      EIP1559Data.maxInclusionFeePerGas,
+      EIP1559Data.maxFeePerGas.sub(baseFee!)
+    )
+    gasPrice = inclusionFeePerGas.add(baseFee!)
+  } else {
```

Comment:
Ok, I have to admit here that I am not able here to completely grasp how an old legacy transaction integrates here on gas price calculation. However can we skip this `getEIP1559Data()` method introduction - which seems very much redundant and unnecessary to me also for an EIP-1559 tx - and instead operate with a `tx.type` switch?

Guess this should be relatively easily doable as well? 🤔

Reply:
(same argument for the second application of this function in `BlockHeader.validateTransactions()`)

Reply:
Yep we can do this, I somewhat got this idea from the EIP which wants to normalize txs, I can add a switch statement. (This does mean we should not forget to add new tx types to this switch statement in case we add new txs)

Reply:
Ok, this was the most delicate fix for me, relatively confident that I translated this correctly, but please have a somewhat careful look on the changes at bb7e3b6e.